sdk/test-utils/recorder-new/src/core-v2-recorder.ts

```diff
@@ -151,6 +170,9 @@ export class TestProxyHttpClient {
             throw new RecorderError("No recording ID returned for a successful start request.");
           }
           this.recordingId = id;
+          if (isPlaybackMode()) {
+            this.variables = !rsp.bodyAsText ? {} : JSON.parse(rsp.bodyAsText);
```

Comment:
```suggestion
            this.variables = JSON.parse(rsp.bodyAsText ?? "{}") as Record<string, string>;
```

Is the body not automatically parsed given the content type?

Reply:
No, that happens in the `deserializationPolicy`

Reply:
Haven't checked,